game modle 'Player' 타입 관련 오류

## 구현할 내용

seeding 완료 이후

```sh
the-pong_backend   | src/api/games/models/game.model.ts:204:7 - error TS2345: Argument of type 'Player' is not assignable to parameter of type 'number'.
the-pong_backend   |
the-pong_backend   | 204       player,
the-pong_backend   |           ~~~~~~
the-pong_backend   |
the-pong_backend   | Found 1 error(s).
the-pong_backend   |
the-pong_backend exited with code 0
```

--- a/backend/src/api/games/dtos/game.dto.ts
+++ b/backend/src/api/games/dtos/game.dto.ts
@@ -140,12 +140,7 @@ export class Game {
   }
 
   hasViewer(tarPlayer: Player): boolean {
-    this.viewers.forEach((player) => {
-      if (player.userId == tarPlayer.userId) {
-        return true;
-      }
-    });
-    return false;
+    return this.viewers.includes(tarPlayer);
   }
 
   getViewerCount(): number {

--- a/backend/src/api/games/models/game.model.ts
+++ b/backend/src/api/games/models/game.model.ts
@@ -201,7 +201,7 @@ export class GameModel implements OnModuleInit {
   async checkUserAchievements(player: Player, gameResult) {
     const userId = player.userId;
     const achievementIds: number[] = await this.getGameAchievements(
-      player,
+      userId,
       gameResult
     );
     const userAchievements = await this.getUserAchievements(userId);

--- a/backend/src/api/games/models/game.model.ts
+++ b/backend/src/api/games/models/game.model.ts
@@ -367,6 +367,23 @@ export class GameModel implements OnModuleInit {
     }
   }
 
+  async checkPlayerInvitation(playerId: number) {
+    const player = this.players.get(playerId);
+    if (!player || !player.gameInvited) return ;
+    const invitedGame = player.gameInvited;
+    const invitetdBy = invitedGame.getOwnerPlayer();
+    if (!invitetdBy) return ;
+
+    await player.socket.emit('gameInvite', {
+      text: 'invited',
+      gameId: invitedGame.gameId,
+      user: {
+        id: invitetdBy.userId,
+        nickname: invitetdBy.username,
+      },
+    });
+  }
+
   async gameStatus(socket: Socket) {
     await socket.emit('gameStatus', {
       games: [...this.games.keys()],

--- a/backend/src/app.gateway.ts
+++ b/backend/src/app.gateway.ts
@@ -127,6 +127,7 @@ export class AppGateway
     }
     if (this.gamesService.gameModel.isPlayerInGame(userId)) {
       this.gamesService.gameModel.reconnectPlayerSocket(userId, socket);
+      this.gamesService.gameModel.checkPlayerInvitation(userId);
       log = `${socket.id} 소켓 재연결 성공 : { id: ${userId}, username: ${username} }`;
     }
     if (!log) {

